<launch>   

    <arg name="lvx_file_path" default="livox_test.lvx"/>
    <arg name="bd_list" default="100000000000000"/>
    <arg name="xfer_format" default="0"/>
    <arg name="multi_topic" default="0"/>
    <arg name="data_src" default="0"/>
    <arg name="publish_freq" default="10.0"/>
    <arg name="output_type" default="0"/>
    <arg name="rviz_enable" default="true"/>
    <arg name="rosbag_enable" default="false"/>
    <arg name="cmdline_arg" default="$(arg bd_list)"/>
    <arg name="msg_frame_id" default="livox_frame"/>
    <arg name="lidar_bag" default="true"/>
    <arg name="imu_bag" default="true"/>

    <param name="xfer_format" value="$(arg xfer_format)"/>
    <param name="multi_topic" value="$(arg multi_topic)"/>
    <param name="data_src" value="$(arg data_src)"/>
    <param name="publish_freq" type="double" value="$(arg publish_freq)"/>
    <param name="output_data_type" value="$(arg output_type)"/>
    <param name="cmdline_str" type="string" value="$(arg bd_list)"/>
    <param name="cmdline_file_path" type="string" value="$(arg lvx_file_path)"/>
    <param name="user_config_path" type="string" value="$(find livox_ros_driver)/config/livox_lidar_config.json"/>	
    <param name="frame_id" type="string" value="$(arg msg_frame_id)"/>
    <param name="enable_lidar_bag" type="bool" value="$(arg lidar_bag)"/>
    <param name="enable_imu_bag" type="bool" value="$(arg imu_bag)"/>
	
	<node name="livox_lidar_publisher" pkg="livox_ros_driver"
	      type="livox_ros_driver_node" required="true"
	      output="screen" args="$(arg cmdline_arg)"/>

    <!-- start PointCloud Fast Filter -->
    <node name="velodyne" pkg="pff" type="velodyne">
        <param name="robot_height" value="0.6"/> <!-- Default 1.0 (100cm) -->
        <param name="sensor_height" value="0.57"/> <!-- Default 0.57 (57cm) -->
        <param name="horizontal_fov" value="360"/> <!-- Default 360째 -->
        <param name="resolution" value="0.2"/> <!-- Default 0.2째 -->
        <param name="topic_pub" value="/scan"/> <!-- Default /scan -->
        <param name="topic_sub" value="/velodyne_points"/> <!-- Default velodyne_points -->
        <param name="frame_id" value="base_link"/> <!-- Default base_scan -->
    </node> 

    <!-- start Turtlebot -->
    <include file="$(find turtlebot_bringup)/launch/minimal.launch"/>

    <arg name="3d_sensor" default="$(env TURTLEBOT_3D_SENSOR)"/>  <!-- r200, kinect, asus_xtion_pro -->


    <!-- Map server -->
    <node name="map_server" pkg="map_server" type="map_server" args="$(find pff_sem)/maps/newmap2.yaml" />

    <!-- AMCL -->
    <arg name="custom_amcl_launch_file" default="$(find turtlebot_navigation)/launch/includes/amcl/amcl.launch.xml"/>
    <!--arg name="initial_pose_x" default="111.92035675"/>
    <arg name="initial_pose_y" default="99.4828338623"/>
    <arg name="initial_pose_a" default="-0.463698840476663"/--><!-- New map-->
    <arg name="initial_pose_x" default="21.3968906403"/>
    <arg name="initial_pose_y" default="7.80321502686"/>
    <arg name="initial_pose_a" default="-0.5812247201566632"/><!-- New map2-->

    <include file="$(arg custom_amcl_launch_file)">
        <arg name="initial_pose_x" value="$(arg initial_pose_x)"/>
        <arg name="initial_pose_y" value="$(arg initial_pose_y)"/>
        <arg name="initial_pose_a" value="$(arg initial_pose_a)"/>
    </include>

    <!-- Move base -->
    <arg name="custom_param_file" default="$(find turtlebot_navigation)/param/$(arg 3d_sensor)_costmap_params.yaml"/>
        <include file="$(find turtlebot_navigation)/launch/includes/move_base.launch.xml">
        <arg name="custom_param_file" value="$(arg custom_param_file)"/>
    </include>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find pff_sem)/rviz_cfg/semantic.rviz"/>

    <!-- start PointCloud people detection -->
    <node name="velodyne_people" pkg="pff_sem" type="velodyne_people">
        <param name="robot_height" value="0.6"/> <!-- Default 1.0 (100cm) -->
        <param name="sensor_height" value="0.57"/> <!-- Default 0.57 (57cm) -->
        <param name="horizontal_fov" value="360"/> <!-- Default 360째 -->
        <param name="resolution" value="0.2"/> <!-- Default 0.2째 -->
        <param name="legs_begin" value="0.25"/> <!-- Default 0.25 (25cm) -->
        <param name="legs_end" value="0.65"/> <!-- Default 0.60 (60cm) -->
        <param name="trunk_begin" value="0.66"/> <!-- Default 0.8 (80cm) -->
        <param name="trunk_end" value="2.00"/> <!-- Default 1.50 (150cm) -->
        
        <param name="topic_pub_people" value="/People_PC"/> <!-- Default "/People_PC" -->
        <param name="topic_pub_legs" value="/Legs_PC"/> <!-- Default "/Legs_PC" -->
        <param name="topic_pub_trunk" value="/Trunk_PC"/> <!-- Default "/Trunk_PC" -->
        <param name="sensor_topic_sub" value="/livox/lidar"/> <!-- Default "/velodyne_points" -->
        <param name="pose_topic_sub" value="/amcl_pose"/> <!-- Default "/amcl_pose" -->
        <param name="map_topic_sub" value="/map"/> <!-- Default "/map" -->
        <param name="frame_id" value="map"/> <!-- Default "map" -->        
    </node> 

    <!-- start PointCloud door detection -->
    <!--node name="velodyne_doors" pkg="pff_sem" type="velodyne_doors">
        <param name="robot_height" value="6.0"/>
        <param name="sensor_height" value="0.57"/>
        <param name="horizontal_fov" value="360"/>
        <param name="resolution" value="0.2"/>
        <param name="topic_pub_doors" value="/Doors_PC"/>
        <param name="sensor_topic_sub" value="/velodyne_points"/>
        <param name="pose_topic_sub" value="/amcl_pose"/>
        <param name="map_topic_sub" value="/map"/>
        <param name="frame_id" value="map"/>
    </node-->

    <!-- start PointCloud path predictor -->
    <!--node name="path_predictor" pkg="pff_sonav" type="path_predictor">
        <param name="topic_pub_path_stored" value="/path_stored"/>
        <param name="topic_pub_path_smoothed" value="/path_smoothed"/>
        <param name="topic_pub_path_predicted" value="/path_predicted"/>
        <param name="sensor_topic_sub" value="/People_PC"/>
        <param name="pose_topic_sub" value="/amcl_pose"/>
        <param name="map_topic_sub" value="/map"/>
        <param name="frame_id" value="map"/>
    </node-->


</launch>
